detectors/gcp/detector_test.go

```diff
@@ -52,6 +52,16 @@ func TestCloudPlatformGKE(t *testing.T) {
 	assert.Equal(t, platform, GKE)
 }
 
+func TestCloudPlatformK8sNotGKE(t *testing.T) {
+	d := NewTestDetector(&FakeMetadataProvider{Err: fmt.Errorf("foo")}, &FakeOSProvider{
+		Vars: map[string]string{
+			k8sServiceHostEnv: "foo",
+		},
+	})
+	platform := d.CloudPlatform()
+	assert.Equal(t, platform, UnknownPlatform)
```

Comment:
It will resolve as being on GCE and detect those attributes.  They would need to detect k8s.cluster.name separately.

It is hard to test the "on GCE, but not on GKE" because the Err applies to all calls.

Reply:
Gotcha, thanks 👍